pkg/result/ignore.go

```diff
@@ -184,6 +184,9 @@ func (c *IgnoreConfig) MatchLicense(licenseID, filePath string) *IgnoreFinding {
 func ParseIgnoreFile(ctx context.Context, ignoreFile string) (IgnoreConfig, error) {
 	var conf IgnoreConfig
 	if _, err := os.Stat(ignoreFile); errors.Is(err, fs.ErrNotExist) {
+		if ignoreFile != DefaultIgnoreFile {
```

Comment:
This check occurs at the end of scanning. We may want to fail on that at an earlier stage. Something like:

```
diff --git a/pkg/flag/report_flags.go b/pkg/flag/report_flags.go
index 67d553b65..d69443e89 100644
--- a/pkg/flag/report_flags.go
+++ b/pkg/flag/report_flags.go
@@ -6,6 +6,7 @@ import (

        "github.com/mattn/go-shellwords"
        "github.com/samber/lo"
+       "github.com/spf13/viper"
        "golang.org/x/xerrors"

        dbTypes "github.com/aquasecurity/trivy-db/pkg/types"
@@ -14,6 +15,7 @@ import (
        "github.com/aquasecurity/trivy/pkg/log"
        "github.com/aquasecurity/trivy/pkg/result"
        "github.com/aquasecurity/trivy/pkg/types"
+       "github.com/aquasecurity/trivy/pkg/utils/fsutils"
        xstrings "github.com/aquasecurity/trivy/pkg/x/strings"
 )

@@ -238,6 +240,10 @@ func (f *ReportFlagGroup) ToOptions() (ReportOptions, error) {
                }
        }

+       if viper.IsSet(f.IgnoreFile.ConfigName) && !fsutils.FileExists(f.IgnoreFile.Value()) {
+               return ReportOptions{}, xerrors.Errorf("ignore file not found: %s", f.IgnoreFile.Value())
+       }
+
        return ReportOptions{
                Format:           format,
                ReportFormat:     f.ReportFormat.Value(),
```

Reply:
Even better !

Thanks for improving my viper knowledge